apps/prairielearn/src/lib/submissions-upload.sql

```diff
@@ -0,0 +1,91 @@
+-- BLOCK select_assessment_question
+SELECT
+  *
+FROM
+  assessment_questions
+WHERE
+  assessment_id = $assessment_id
+  AND question_id = $question_id;
+
+-- BLOCK insert_assessment_instance
+INSERT INTO
+  assessment_instances (assessment_id, user_id, number, open)
+VALUES
+  (
+    $assessment_id,
+    $user_id,
+    $instance_number,
+    false -- Assume closed by default when recreating
+  )
+RETURNING
+  id AS assessment_instance_id;
+
+-- BLOCK insert_instance_question
+INSERT INTO
+  instance_questions (
+    assessment_instance_id,
+    assessment_question_id,
+    requires_manual_grading,
+    status
+  )
+VALUES
+  (
+    $assessment_instance_id,
+    $assessment_question_id,
+    $requires_manual_grading,
+    'saved' -- Must not be the default 'unanswered' status
+  )
+RETURNING
+  id AS instance_question_id;
+
+-- BLOCK insert_variant
+INSERT INTO
+  variants (
+    instance_question_id,
+    question_id,
+    authn_user_id,
+    user_id,
+    variant_seed,
+    params,
+    true_answer,
+    options,
+    course_id
+  )
+VALUES
+  (
+    $instance_question_id,
+    $question_id,
+    $authn_user_id,
+    $user_id,
+    $seed,
+    $params,
+    $true_answer,
+    $options,
+    $course_id
+  )
+RETURNING
+  id AS variant_id;
+
+-- BLOCK insert_submission
+INSERT INTO
+  submissions (
+    variant_id,
+    auth_user_id,
+    submitted_answer,
+    raw_submitted_answer,
+    params,
+    true_answer,
+    date
+  )
+VALUES
+  (
+    $variant_id,
+    $authn_user_id,
+    $submitted_answer,
+    '{}'::jsonb, -- We don't have any useful value for `raw_submitted_answer` here.
```

Comment:
Minor issue when this isn't set: some elements (e.g., pl-integer-input, pl-string-input) use the raw submitted answer to show the value in the question panel. Not setting it here causes these elements not to have a value to show in that panel. I don't think it's a problem, just making sure you're aware of it.

Reply:
Yeah, I had noticed this. I think this can be addressed in the future if/when we decide to include `raw_submitted_answer` in the CSV download.